Tools/TestWebKitAPI/Tests/WebCore/FontSizeAdjust.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2023 Igalia, S.L. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+
+#include <WebCore/FontSizeAdjust.h>
+
+namespace TestWebKitAPI {
+
+using namespace WebCore;
+
+TEST(FontSizeAdjustTest, Metric)
+{
+    FontSizeAdjust fontSizeAdjust;
```

Comment:
The FontSizeAdjust struct is initialized via the list initialization when it is created. There is no declaration or use like this in the current code IIRC. Thus, we got no style issue.
By the way, this test just checks if an uninitialized struct has a garbage value or not. As others said, it might be better to have a web test showing a problem in practice instead...

Reply:
I've updated the test. It should now better illustrate the problem. The new test basically simulates this logic:
https://github.com/WebKit/WebKit/blob/3b9112ff64a6aa1c0daf62390d1646bcc36b488b/Source/WebCore/dom/Document.cpp#L2142-L2145
It is hard to write a web test for it because the issue is that the style update will be propagated starting from `renderView` down the tree. But this can hide other problems like #12312 which is now only reproducible on Linux, probably because of using `gcc` instead of `clang`.